kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java

```diff
@@ -72,74 +57,61 @@
-                                                          "localhost");
-    private static final int PORT = Integer.parseInt(System.getProperty(ROUTER_PORT,
-                                                                        String.valueOf(DEFAULT_PORT_NUM)));
-    private static final int PORT_TLS = Integer.parseInt(System.getProperty(ROUTER_PORT_TLS,
-                                                                            String.valueOf(DEFAULT_PORT_TLS_NUM)));
-    private static final String KEYSTORE_PATH = System.getProperty(ROUTER_KEYSTORE);
-    private static final String KEYSTORE_PASSWORD = System.getProperty(ROUTER_KEYSTORE_PASSWORD);
-    private static final String KEYSTORE_KEYALIAS = System.getProperty(ROUTER_KEYSTORE_KEYALIAS);
-    private static final boolean TLS_ENABLED = KEYSTORE_PATH != null && !KEYSTORE_PATH.isEmpty();
-    private int failedAttemptsInterval = Integer.parseInt(System.getProperty(KIE_SERVER_CONTROLLER_ATTEMPT_INTERVAL, "10"));
-
-    private static boolean MANAGEMENT_SECURED = isManagementSecured();
-    private static String IDENTITY_PROVIDER = getIdentityProvider();
-
     public static final String CMD_ADD_USER = "addUser";
     public static final String CMD_REMOVE_USER = "removeUser";
 
-    private String CONTROLLER = System.getProperty(KIE_CONTROLLER);
-
     private static final Logger log = Logger.getLogger(KieServerRouter.class);
 
-    private static final String SERVER_INFO_JSON = "{\n" +
-            "      \"version\" : \"LATEST\",\n" +
-            "      \"name\" : \"" + KieServerInfoHandler.getRouterName() + "\",\n" +
-            "      \"location\" : \"" + KieServerInfoHandler.getLocationUrl() + "\",\n" +
-            "      \"capabilities\" : [ \"KieServer\", \"BRM\", \"BPM\", \"CaseMgmt\", \"BPM-UI\", \"BRP\" ],\n" +
-            "      \"id\" : \"" + KieServerInfoHandler.getRouterId() + "\"\n" +
-            "}";
-
-
     private ServiceLoader<ConfigRepository> configRepositoryServiceLoader = ServiceLoader.load(ConfigRepository.class);
 
     private Undertow server;
-    private ConfigRepository repository = new FileRepository();
+    private ConfigRepository repository;
 
     private ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
     private ScheduledFuture<?> controllerConnectionAttempts;
-    private boolean isSecured;
-    private String identityServiceName;
-    private boolean isHttpEnabled;
+    private KieServerRouterEnvironment env;
 
     public KieServerRouter() {
-        this(reloadManagementSecured(), reloadIdentityProvider()); //Reload for easier testing
+        this(new KieServerRouterEnvironment());
     }
 
-    public KieServerRouter(boolean isSecured, String identityServiceName) {
+    public KieServerRouter(KieServerRouterEnvironment env) {
         configRepositoryServiceLoader.forEach(repo -> repository = repo);
         log.info("KIE Server router repository implementation is " + repository);
```

Comment:
Should be placed after `this.repository = new FileRepository(env);`

Reply:
done